x-pack/test_serverless/api_integration/test_suites/observability/apm_api_integration/feature_flags.ts

```diff
@@ -0,0 +1,223 @@
+        } catch (err) {
+          expect(err.res.status).toBe(agentConfigurationResponse.statusCode);
+          expect(err.res.body).toStrictEqual(agentConfigurationResponse);
+        }
+      });
+
+      // Not possible to test the DELETE, as first we will need to get the service
+      xit('rejects requests to delete agent configuration', async () => {
+        try {
+          await apmApiClient.slsUser({
+            endpoint: 'DELETE /api/apm/settings/agent-configuration 2023-10-31',
+            params: {
+              body: {
+                service: {},
+                settings: { transaction_sample_rate: '0.55' },
+              },
+            },
+          });
+        } catch (err) {
+          expect(err.res.status).toBe(agentConfigurationResponse.statusCode);
+          expect(err.res.body).toStrictEqual(agentConfigurationResponse);
+        }
+      });
+
+      it('rejects requests to create/update agent configuration', async () => {
+        try {
+          await apmApiClient.slsUser({
+            endpoint: 'PUT /api/apm/settings/agent-configuration 2023-10-31',
+            params: {
+              body: {
+                service: {},
+                settings: { transaction_sample_rate: '0.55' },
+              },
+            },
+          });
+        } catch (err) {
+          expect(err.res.status).toBe(agentConfigurationResponse.statusCode);
+          expect(err.res.body).toStrictEqual(agentConfigurationResponse);
+        }
+      });
+
+      it('rejects requests to lookup single configuration', async () => {
+        try {
+          await apmApiClient.slsUser({
+            endpoint: 'POST /api/apm/settings/agent-configuration/search 2023-10-31',
+            params: {
+              body: {
+                service: { name: 'myservice', environment: 'development' },
+                etag: '7312bdcc34999629a3d39df24ed9b2a7553c0c39',
+              },
+            },
+          });
+        } catch (err) {
+          expect(err.res.status).toBe(agentConfigurationResponse.statusCode);
+          expect(err.res.body).toStrictEqual(agentConfigurationResponse);
+        }
+      });
+    });
+
+    describe('source maps', () => {
+      it('rejects requests to list source maps', async () => {
+        try {
+          await apmApiClient.slsUser({
+            endpoint: 'GET /api/apm/sourcemaps 2023-10-31',
+          });
+        } catch (err) {
+          expect(err.res.status).toBe(sourceMapsResponse.statusCode);
+          expect(err.res.body).toStrictEqual(sourceMapsResponse);
+        }
+      });
+
+      it('rejects requests to upload source maps', async () => {
+        try {
+          await uploadSourcemap(apmApiClient);
+        } catch (err) {
+          expect(err.res.status).toBe(sourceMapsResponse.statusCode);
+          expect(err.res.body).toStrictEqual(sourceMapsResponse);
+        }
+      });
+      // Not possible to test the DELETE, as first we will need to get the id of a source map
```

Comment:
Can't we create a sourcemap first as we do in https://github.com/elastic/kibana/blob/50a9e13035037694b379d5811029497bc264f2f2/x-pack/test/apm_api_integration/tests/sourcemaps/sourcemaps.ts#L308?